src/ray/raylet/scheduling/cluster_resource_data.cc

```diff
@@ -58,9 +58,14 @@ float NodeResources::CalculateCriticalResourceUtilization() const {
     if (total == 0) {
       continue;
     }
-    const auto &available = this->available.Get(ResourceID(i));
-
-    float utilization = 1 - (available.Double() / total.Double());
+    auto available = this->available.Get(ResourceID(i)).Double();
+    if (this->normal_task_resources.Has(ResourceID(i))) {
+      available -= this->normal_task_resources.Get(ResourceID(i)).Double();
+      if (available < 0) {
+        available = 0;
+      }
+    }
```

Comment:
Why do we need this?

Reply:
As we mentioned above, gcs scheduler handles normal_task_resources specifically. So when calculating the available resources, we have to take one more step to take that into account. Besides, for raylet scheduling, the `normal_task_resources` is always empty, so don't have to worry about it here.

Reply:
Please write this comment to the code block (it is not straightforward to understand )